### DISCUSSION TITLE
Dropping python 2.7 support

### DISCUSSION THREAD

**Discussion by alejandroautalan** [alejandroautalan/pygubu-designer#75] [Q&A] [answered]
Hello all.
I'm working on plugins to allow pygubu designer to use widgets from external projects.

Some of them are:
- ttkwidgets
- ttkbootstrap (wishlist)

The latest versions of this projects support python3 only (and some python3 specific versions).

So I want to clean the codebase and also maybe, jump to a minimal python3 version.

Is there a minimal python version that you people recommend?

Regards
Alejandro A.

**Comment by rdbende** [comment]
Actually for my own projects I wouldn't support python versions older than 3.7, but I know that there are people using Pygubu with much older Python versions, so 3.7 wouldn't be a good choice.
I don't think Pygubu should support versions older than 3.4, and the code would be much more modern this way. With 3.4 you can also use the pathlib module, which is fantastic!

**Comment by jrezai** [comment]
Hello,

Similar to @rdbende, I also wouldn't use any version older than 3.7 for my own projects. One big milestone, in my opinion, was Python 3.6 which introduced f-strings (v3.6 came out in 2016). I use f-strings regularly. So my vote would be for at least Python 3.6 because of f-strings. However, I would be fine with v3.4 too.

Thanks,

**Comment by alejandroautalan** [comment, marked as answer]
Hello @rdbende and @jrezai, thanks for your comments.
I also like libpath and fstrings. I'll do some more research but I think version 3.6 will be the one to choose.

Thanks!
Alejandro A.
